src/parallel/CollectiveCommunicationSingleNonBlocking.h

```diff
@@ -149,12 +146,13 @@ class CollectiveCommunicationSingleNonBlocking: public CollectiveCommunication {
 
 	}
 
-	MPI_Request* _request;
-	MPI_Op _agglomeratedTypeAddOperator;
-	bool _communicationInitiated;
-	bool _valuesValid;
+	std::unique_ptr<MPI_Request> _request{new MPI_Request()};
+	MPI_Op _agglomeratedTypeAddOperator{MPI_OP_NULL};
+	bool _communicationInitiated{false};
+	bool _valuesValid{false};
+	/// tempValues is used for overlapped communications!
 	std::vector<valType> _tempValues;
-	bool _firstComm;
+	bool _firstComm{true};
 };
 
 #endif // MPI_VERSION >= 3
```

Comment:
Is this documented somewhere that this need MPI > 3 ? I would expect at least a warning at compile time.

Reply:
non-blocking collectives are known to require mpi >= 3. or aren't they?

Reply:
Added it to the all-options.xml and the doc of the CollectiveCommunicationNonBlocking class
There's already a warning if it is set in the XML, but not compiled: https://github.com/ls1mardyn/ls1-mardyn/blob/ae4cefceb4dd50d8809bdbb73941fca86b3a7f0e/src/parallel/DomainDecompMPIBase.cpp#L124